plotly/tools.py

```diff
@@ -3442,3 +3694,226 @@ def get_dendrogram_traces(self, X, colorscale):
             trace_list.append(trace)
 
         return trace_list, icoord, dcoord, ordered_labels, P['leaves']
+
+
+class _AnnotatedHeatmap(FigureFactory):
+    """
+    Refer to TraceFactory.create_annotated_heatmap() for docstring
+    """
+    def __init__(self, z, x, y, annotation_text, colorscale,
+                 fontcolor, reversescale, **kwargs):
+        from plotly.graph_objs import graph_objs
+
+        self.z = z
+        if x:
+            self.x = x
+        else:
+            self.x = range(len(z[0]))
+        if y:
+            self.y = y
+        else:
+            self.y = range(len(z))
+        if annotation_text is not None:
+            self.annotation_text = annotation_text
+        else:
+            self.annotation_text = self.z
+        self.colorscale = colorscale
+        self.reversescale = reversescale
+        self.fontcolor = fontcolor
+
+    def get_text_color(self):
+        '''
+        Get font color for annotations.
+
+        The annotated heatmap can feature 2 text colors: min_text_color and
```

Comment:
:cow2: `two` > `2` here I think. Feel free to ignore this very, very small :goat: